client/src/profile/ProfileContainer.js

```diff
@@ -1,50 +1,115 @@
-import React from 'react';
+import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
-import ChangePasswordForm from './ChangePasswordForm';
+import UserForm from './UserForm';
 import ProfileForm from './ProfileForm';
-import AddressField from './AddressField';
+import Addresses from './Addresses';
+import { getProfile, updateProfile } from './profileActions';
 
 class ProfileContainer extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
       updatingPassword: false,
+      newAddressState: '',
     };
 
     this.submitUpdate = this.submitUpdate.bind(this);
     this.togglePasswordForm = this.togglePasswordForm.bind(this);
+    this.saveNewAddress = this.saveNewAddress.bind(this);
+    this.updateAddressState = this.updateAddressState.bind(this);
+    this.removeAddress = this.removeAddress.bind(this);
+    this.updatePassword = this.updatePassword.bind(this);
+  }
+  componentWillMount() {
+    this.props.getProfile();
   }
   submitUpdate(values) {
-    console.log(this.props);
-    console.log(values);
+    const newProfile = Object.assign({}, this.props.profile, values);
+    this.props.updateProfile(newProfile);
   }
   togglePasswordForm(e) {
     e.preventDefault();
     this.setState({ updatingPassword: !this.state.updatingPassword });
   }
+  updateAddressState(newState) {
+    this.setState({ newAddressState: newState });
+  }
+  removeAddress(address) {
+    const newProfile = Object.assign({}, this.props.profile);
+    const loc = newProfile.addresses.indexOf(address);
+    if (loc === -1) {
+      console.log('Attempting to remove an address that is not in the list');
```

Comment:
This error is a bug in our code, not something that a user can reasonably recover from. In objc I would assert and crash the app.